File: python/sentroy/domains.py
from __future__ import annotations

import urllib.parse

from sentroy._http import _HttpClient
from sentroy.types import Domain


class DomainsResource:
    """Interact with the Domains API."""

    def __init__(self, http: _HttpClient) -> None:
        self._http = http

    def list(self) -> list[Domain]:
        """List all verified domains for the company."""
        data = self._http.get("/domains")
        return [Domain.from_dict(d) for d in (data or [])]

    def get(self, id: str) -> Domain:
        """Get a single domain by ID."""
        data = self._http.get(f"/domains/{urllib.parse.quote(id, safe='')}")
        return Domain.from_dict(data)

File: typescript/src/resources/domains.ts
import type { HttpClient } from "../http"
import type { Domain } from "../types"

export class Domains {
  constructor(private http: HttpClient) {}

  /** List all verified domains for the company */
  async list(): Promise<Domain[]> {
    return this.http.get<Domain[]>("/domains")
  }

  /** Get a single domain by ID */
  async get(id: string): Promise<Domain> {
    return this.http.get<Domain>(`/domains/${encodeURIComponent(id)}`)
  }
}
